src/edge_utils.c

```diff
@@ -275,6 +266,28 @@ int supernode_connect(n2n_edge_t *eee) {
             traceEvent(TRACE_INFO, "PMTU discovery %s", (eee->conf.disable_pmtu_discovery) ? "disabled" : "enabled");
 #endif
 
+        // detetct local port even/especially if chosen by OS...
+        if((getsockname(eee->sock, (struct sockaddr *)&local_sock, &sock_len) == 0)
+        && (local_sock.sin_family == AF_INET)
+        && (sock_len == sizeof(local_sock))) {
+            // ... and write to local preferred socket -- no matter if used or not
+            eee->conf.preferred_sock.port = ntohs(local_sock.sin_port);
+            // probe for & overwrite local address only if auto-detection mode
+            if(eee->conf.preferred_sock_auto) {
+                probe_sock = socket(PF_INET, SOCK_DGRAM, 0);
+                // connecting the UDP socket makes getsockname read the local address it uses to connect (to the sn in this case)
+                // we cannot do it with the real (eee->sock) socket because socket does not accept any conenction from elsewhere then,
+                // e.g. from another edge instead of the supernode; hence, we use a temporary socket
+                connect(probe_sock, (struct sockaddr *)&sn_sock, sizeof(sn_sock));
```

Comment:
Please take the acquisition of the local IP address as an independent public function, which is also needed in other places, using `getsockname` to get the local address is very concise. I think it is possible to delete the `get_best_interface_ip` in `src/edge_utils_win32.c` and replace it.

Reply:
Can do. Not sure about the windows part though.

Reply:
#781